### PR TITLE
[LW-9213] update cardano-sdk packages used in lace

### PR DESCRIPTION
**Pull request by mchappell**
# Checklist

- [ ] JIRA - [LW-9213](https://input-output.atlassian.net/browse/LW-9213)

## Testing

Full e2e run against this branch

[LW-9213]: https://input-output.atlassian.net/browse/LW-9213?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mkazlauskas — APPROVED** [review]
LGTM

**Review by mirceahasegan — APPROVED** [review]

**Comment by mchappell** [comment]
Do not merge, this PR will require deployments to be done to live environments

File: packages/cardano/src/wallet/lib/hardware-wallet.ts [hunk 1]
```diff
@@ -175,43 +107,14 @@ export const createHardwareWalletsByChain = async (
...
-                trezorConfig: TREZOR_CONFIG
-              },
-              dependencies
-            );
-          },
-          createWallet,
-          logger: console,
-          bip32Ed25519: new Crypto.CmlBip32Ed25519(CML)
-        });
+        const { keyAgent } = await await setup({ chainId, connectedDevice });
```
Comment (pczeglik-iohk) [inline packages/cardano/src/wallet/lib/hardware-wallet.ts:110]: Is this `await await` needed? Sonar complains :) https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=761&id=input-output-hk_lace&open=AYxI1yIe5eCjlJW9M8IQ

File: packages/cardano/src/wallet/test/mocks/StakepoolSearchProviderStub.ts [hunk 2]
```diff
@@ -20,7 +20,9 @@ export const pools: Partial<Cardano.StakePool>[] = [
       livePledge: BigInt('2000000000'),
       saturation: Percent(0.211),
       size: undefined,
-      stake: undefined
+      stake: undefined,
+      ros: Percent(0.69),
```
Comment (pczeglik-iohk) [inline packages/cardano/src/wallet/test/mocks/StakepoolSearchProviderStub.ts:24]: Sonar complains as well here as it spots this as a code duplication: https://sonarcloud.io/component_measures?metric=new_duplicated_lines_density&selected=input-output-hk_lace%3Apackages%2Fcardano%2Fsrc%2Fwallet%2Ftest%2Fmocks%2FStakepoolSearchProviderStub.ts&view=list&pullRequest=761&id=input-output-hk_lace

File: apps/browser-extension-wallet/src/features/unlock-wallet/components/UnlockWalletContainer.tsx [hunk 3]
```diff
@@ -66,6 +66,7 @@ export const UnlockWalletContainer = ({ validateMnemonic }: UnlockWalletContaine
       setIsValidPassword(true);
       analytics.sendEventToPostHog(PostHogAction.UnlockWalletWelcomeBackUnlockClick);
       if (decrypted) setUnlocked(decrypted);
+      setAddressesDiscoveryCompleted(true);
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/features/unlock-wallet/components/UnlockWalletContainer.tsx:69]: I am not sure about setting this flag here... This flag is managed by the wlletManager's `updateAddresses` method which gets triggered on a message from `wallet.addresses$` which I assume arrives after unlocking. The `updateAddresses` checks a few things, ensures consistency of stored addresses, etc. and only there sets this flag.

**Pull request merged**